core/src/eval/fixpoint.rs

```diff
@@ -54,21 +44,18 @@ fn patch_term<C: Cache>(
 pub fn rec_env<'a, I: Iterator<Item = (&'a LocIdent, &'a Field)>, C: Cache>(
     cache: &mut C,
     bindings: I,
-    env: &Environment,
     pos_record: TermPos,
-) -> Result<Vec<(Ident, CacheIndex)>, EvalError> {
+) -> Vec<(Ident, CacheIndex)> {
     bindings
         .map(|(id, field)| {
             if let Some(ref value) = field.value {
                 let idx = match value.as_ref() {
-                    Term::Var(ref var_id) => env
-                        .get(&var_id.ident())
-                        .cloned()
-                        .ok_or(EvalError::UnboundIdentifier(*var_id, value.pos))?,
+                    Term::Closure(idx) => idx.clone(),
                     _ => {
                         // If we are in this branch, `rt` must be a constant after the share normal
                         // form transformation, hence it should not need an environment, which is
```

Comment:
What happens here now that the share normal form doesn't exist anymore?

Reply:
It is now enforced by the eval loop itself. I'll update this comment.